src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1517,7 +1534,34 @@ namespace winrt::TerminalApp::implementation
                     text = item.Path();
                 }
             }
-            eventArgs.HandleClipboardData(text);
+
+            co_await winrt::resume_foreground(Dispatcher());
+
+            const bool hasNewLine = std::find(text.cbegin(), text.cend(), L'\n') != text.cend();
+            const bool warnMultiLine = hasNewLine && _settings->GlobalSettings().WarnAboutMultiLinePaste();
+
+            constexpr const std::size_t minimumSizeForWarning = 1024 * 5; // 5 KiB
```

Comment:
I put 5 KiB here for now, but I don't know what should be the _optimal_ threshold for the warning.

Reply:
I think five kilobytes is fine!